### ISSUE TITLE
Okteto up not working anymore on AWS

### ISSUE THREAD

**Issue by AnesBenmerzoug** [okteto/okteto#506]
**Describe the bug**

After upgrading to version 1.5.1 from version 1.4.9 ( because of the custom annotations ) okteto gets stuck while activating the development environment and then errors with the following message:
```
 x  error creating PVC 'okteto'. It is 'Pending' after 150s
```

**To Reproduce**
Steps to reproduce the behaviour:
1. Upgrade to version 1.5.1
2. Run 
```bash
$ okteto up
```
3. Wait for a bit
```
Installing dependencies...
syncthing-Linux-x86_64 20.30 MiB / 20.30 MiB [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 8.07 MiB p/s
Deployment nvoffice-nvo doesn't exist in namespace dev. Do you want to create a new one? [y/n]: y
Activating your development environment... 
```
4. Error happens
```
 x  error creating PVC 'okteto'. It is 'Pending' after 150s
```

**Expected behavior**
With version 1.4.9 everything works fine.
Except when trying to go back into the environment. For that I just delete the okteto statefulset and run okteto up again

**Desktop:**
 - OS: Ubuntu 18.04
 - Version 18.04

**Additional context**
I tried this only on AWS

[okteto.log](https://github.com/okteto/okteto/files/3767506/okteto.log)

**Comment by pchico83** [comment]
@AnesBenmerzoug I am trying to repro the issue with no luck for the moment.
Are you using a multi-zone cluster? does your deployment have other persistent volumes attached?
Could you check why the volumen is in `Pending` state?

**Comment by AnesBenmerzoug** [comment]
@pchico83 Hmm, that's strange.
I am using an EKS cluster that spans 3 zones. Would that create a problem?
Yes, there are other persistent volumes attached and they're working fine.
I tried to find out why that particular volume was stuck, but I couldn't find anything.
Forgot to add that I'm using Kubernetes version 1.14.7, if that is of any use.

**Comment by pchico83** [comment]
@AnesBenmerzoug could it be that the okteto volume is created in a different availability zone than the ones in your deployment?
If that is the issue, what is the `volumeBindingMode` of your default storage class?
You can define `volumeBindingMode: WaitForFirstConsumer` in your default storage class to overcome this problem. `WaitForFirstConsumer` means volumes are not immediately created. Instead, the creation is delayed until a pod mounts the volume, ensuring it is created in the right zone.

**Comment by AnesBenmerzoug** [comment]
@pchico83 Hmmm, I didn't think about that.
when I deploy the okteto development environment a persistent volume with default class is created.
I think it has by default an immediate binding mode.
I remember I tried to set a different storage class for okteto, but I couldn't find an option for it.
I will try tomorrow to check in which zone the volume is created.

**Comment by pchico83** [comment]
@AnesBenmerzoug we don't support specifying the storage class at this moment. I just filed an issue for it: https://github.com/okteto/okteto/issues/508

If you confirm this is the issue, we can pack a new release for next monday implementing storage class support.

**Comment by pchico83** [comment]
@AnesBenmerzoug would you find it useful to have an option to deactivate PV creation for `okteto up`?
PVs are useful to persist programming language caches, dependencies installation like node modules, online IDE extensions ... but we have also considered having an option to deactivate PV creation and make the deployment of the dev environment less dependent on the cluster topology.

**Comment by AnesBenmerzoug** [comment]
@pchico83 sorry the late reply. Itt was a busy day at work.
I changed the default storage class and set the binding mode to waitForFirstConsumer, but nothing changed. It still gets stuck.
Being able to specify the storage class and to the PVs optional would be a nice addition.
I will see if I can pinpoint the specific commit where things break.

**Comment by pchico83** [comment]
@AnesBenmerzoug just to double-check, did you delete the `okteto` PV before trying the waitForFirstConsumer binding mode?
We are working on making PVs optional and storage class support in the meantime.

**Comment by AnesBenmerzoug** [comment]
@pchico83 Yes, I made sure to delete everything before trying the waitForFirstConsumer binding mode.

This is the storage class I used:

```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  labels:
    stack: backend
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
  name: default-persistent-storage
provisioner: kubernetes.io/aws-ebs
parameters:
  type: gp2
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
mountOptions:
- debug
allowVolumeExpansion: true
```

This what I get when I describe the PVC. It is stuck waiting for the first consumer.

```yaml
Name:          okteto
Namespace:     dev
StorageClass:  default-persistent-storage
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   <none>
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Events:
  Type       Reason                Age                  From                         Message
  ----       ------                ----                 ----                         -------
  Normal     WaitForFirstConsumer  9s (x13 over 2m42s)  persistentvolume-controller  waiting for first consumer to be created before binding
Mounted By:  <none>
```

**Comment by pchico83** [comment]
@AnesBenmerzoug thanks for your answer and detailed logs. I was looking at the wrong place, I think I have identified the issue, fix is coming.
Also, we will work on storage classes and non-persistent volumes in following PRs.

**Comment by pchico83** [comment]
@AnesBenmerzoug could you try with the latest bits from master?
If it works, we can pack a release later tonight.

**Comment by AnesBenmerzoug** [comment]
@pchico83 I just tested and it's finally working. Thanks for the great work!
From what I understood, previously okteto would get stuck waiting for the PV and error out before even start the pod whereas now it just creates the PVC and moves on. Is that correct?

**Comment by pchico83** [comment]
Right. I thought the issue was that the PVC was created in a different zone, but you were already using `WaitForFirstConsumer`. I have created issues to improve our support for multi-zone clusters, but I think we can close this issue for now.
Thanks for reporting this regression :-)

**Comment by pchico83** [comment]
Closed by #513
